### PR TITLE
Ea validate annotation files

### PR DESCRIPTION
**Pull request by knapii-developments**
Due to sentry errors seen in metadata and cluster files, additional formatting checks have been added:

- metadata file validator confirms there are no x, y, or z columns
- cluster files validator confirms no missing "X" and "Y" columns 

Tests have been added for mentions above. 

This PR satifies [SCP-2139](https://broadworkbench.atlassian.net/browse/SCP-2139) and [SCP-2338](https://broadworkbench.atlassian.net/browse/SCP-2338)

### CODE REVIEW

File: ingest/clusters.py [hunk 1]
```diff
@@ -75,6 +76,32 @@ def __init__(
...
+    def validate_header_for_coordinate_values(self):
+        """Cluster files must have coordinates 'x' and 'y' in header
+        :return: boolean True if coordinates are in header, otherwise False
+        """
+        lower_cased_headers = [header.lower() for header in self.headers]
+        for coordinate in ('x', 'y'):
+            if coordinate not in lower_cased_headers:
+                msg = (
+                    f"Header must have coordinate values 'x' and 'y' (case insensitive)"
+                )
```
Comment (eweitz) [inline ingest/clusters.py:100]: No interpolation is being done here, so an f-string seems unnecessary -- a basic string seems simpler and sufficient.  

(I suspect this syntax noise comes from a linter -- and that refining linter settings would fix it.)

```suggestion
                msg = (
                    "Header must have coordinate values 'x' and 'y' (case insensitive)"
                )
```

File: ingest/cell_metadata.py [hunk 2]
```diff
@@ -60,6 +61,33 @@ class Model(TypedDict):
...
+        :return: boolean True if coordinates are not in header, otherwise False
+        """
+        lower_cased_headers = [header.lower() for header in self.headers]
+        valid = not any(
+            [coordinate in ('x', 'y', 'z') for coordinate in lower_cased_headers]
+        )
+        if valid:
+            return True
+        else:
+            msg = f'Header names can not be coordinate values x, y, or z (case insensitive)'
```
Comment (eweitz) [inline ingest/cell_metadata.py:87]:
```suggestion
            msg = 'Header names can not be coordinate values x, y, or z (case insensitive)'
```

File: ingest/ingest_pipeline.py [hunk 3]
```diff
@@ -405,7 +405,8 @@ def ingest_expression(self) -> int:
     # @my_debug_logger()
     def ingest_cell_metadata(self):
         """Ingests cell metadata files into Firestore."""
-        if self.cell_metadata.validate_format():
+        if self.cell_metadata.validate():
+            pass
```
Comment (eweitz) [inline ingest/ingest_pipeline.py:409]: This `pass` seems extraneous -- can it be removed?

```suggestion
```

File: tests/test_cell_metadata.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import unittest
+
+sys.path.append("../ingest")
+from cell_metadata import CellMetadata
+
+
+class TestCellMetadata(unittest.TestCase):
+    def test_validate_header_for_coordinate_values_false(self):
+        """Ensures validate_header_for_coordinate_values returns false when
+        coordintate value is in metadata file
```
Comment (eweitz) [inline tests/test_cell_metadata.py:11]:
```suggestion
        coordinate value is in metadata file
```

And similarly throughout this file.

File: ingest/annotations.py [hunk 5]
```diff
@@ -240,9 +240,9 @@ def validate_against_header_count(self):
         return valid
 
     def validate_format(self):
-        """Check all metadata file format criteria for file validity
+        """Check common format criteria for annotation files
         """
-        self.is_valid_file = all(
+        return all(
```
Comment (devonbush) [inline ingest/annotations.py:245]: nice cleanup

File: ingest/cell_metadata.py [hunk 6]
```diff
@@ -35,7 +36,7 @@ def __init__(
         study_id: ObjectId,
         study_file_id: ObjectId,
         *args,
-        **kwargs
+        **kwargs,
```
Comment (devonbush) [inline ingest/cell_metadata.py:39]: I learned something new -- I see this is recommended by PEP8 for multiline lists, nice

File: tests/test_cell_metadata.py (hunk 4 above)
Comment (knapii-developments) [reply tests/test_cell_metadata.py]: 0f2b689

File: ingest/ingest_pipeline.py (hunk 3 above)
Comment (knapii-developments) [reply ingest/ingest_pipeline.py]: 0f2b689

File: ingest/cell_metadata.py (hunk 2 above)
Comment (knapii-developments) [reply ingest/cell_metadata.py]: 0f2b689

File: ingest/clusters.py (hunk 1 above)
Comment (knapii-developments) [reply ingest/clusters.py]: 0f2b689

**Pull request merged**